test/connectors/test_torch_connector.py

```diff
@@ -568,6 +572,65 @@ def test_batch_gradients(self):
             places=4,
         )
 
+    @data(
+        # output_shape, interpret
+        (1, None),
+        (2, lambda x: "{:b}".format(x).count("1") % 2),
+    )
+    @requires_extra_library
+    def test_circuit_qnn_batch_gradients(self, config):
+        """Test batch gradient computation of CircuitQNN gives the same result as the sum of
+        individual gradients."""
+
+        output_shape, interpret = config
+        num_inputs = 2
+
+        feature_map = ZZFeatureMap(num_inputs)
+        ansatz = RealAmplitudes(num_inputs, entanglement="linear", reps=1)
+
+        qc = QuantumCircuit(num_inputs)
+        qc.append(feature_map, range(num_inputs))
+        qc.append(ansatz, range(num_inputs))
+
+        qnn = CircuitQNN(
+            qc,
+            input_params=feature_map.parameters,
+            weight_params=ansatz.parameters,
+            interpret=interpret,
+            output_shape=output_shape,
+            quantum_instance=self.sv_quantum_instance,
+        )
+
+        # set up PyTorch module
+        initial_weights = np.array([0.1] * qnn.num_weights)
+        model = TorchConnector(qnn, initial_weights)
+
+        # random data set
+        x = Tensor(np.random.rand(5, 2))
+        y = Tensor(np.random.rand(5, output_shape))
+
+        # define optimizer and loss
+        optimizer = SGD(model.parameters(), lr=0.1)
+        f_loss = MSELoss(reduction="sum")
+
+        sum_of_individual_losses = 0.0
+        for x_i, y_i in zip(x, y):
+            output = model(x_i)
+            sum_of_individual_losses += f_loss(output, y_i)
+        optimizer.zero_grad()
+        sum_of_individual_losses.backward()
+        sum_of_individual_gradients = deepcopy(model.weights.grad)
+
+        output = model(x)
+        batch_loss = f_loss(output, y)
+        optimizer.zero_grad()
+        batch_loss.backward()
```

Comment:
An additional equality assert of `sum_of_individual_losses` and `batch_loss` would be great.

Reply:
that's happening on lines 630-632. is that not sufficient?

Reply:
on these line you compare gradients, not loss:
```python
        self.assertAlmostEqual(
            np.linalg.norm(sum_of_individual_gradients - batch_gradients), 0.0, places=4
        )
```
I suggest to add a comparison of loss as well since both values are already computed, something like:
```python
    self.assertAlmostEqual(sum_of_individual_losses, batch_loss)
```

Reply:
ah yes, you're right - my bad